resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java

```diff
@@ -347,10 +352,71 @@ public Tuple2<ResponseWrapper<VidResponseDto>, String> generateVidV2(BaseVidRequ
 			responseDto.setVersion(version);
 		}
 		responseDto.setResponsetime(DateUtils.formatToISOString(DateUtils.getUTCCurrentDateTime()));
-
 		return Tuples.of(responseDto, eventId);
 	}
 
+	private void validateVidFromSession(String individualId) {
+		try {
+			String idType = identityServiceImpl.getIndividualIdType(individualId);
+			String uin = identityServiceImpl.getUinForIndividualId(individualId);
+			if (idType.equalsIgnoreCase(IdType.VID.name())) {
+				String vidType =
+						getVidTypeFromVid(individualId, uin);
+				if(vidType.equalsIgnoreCase(ResidentConstants.PERPETUAL)){
+					int numberOfPerpetualVid = getNumberOfPerpetualVidFromUin(uin).getT1();
+					VidPolicy vidPolicy = getVidPolicyAsVidPolicyDto();
+					if(vidPolicy.getAllowedInstances() >= numberOfPerpetualVid
+							&& vidPolicy.getAutoRestoreAllowed() &&
+							!vidPolicy.getRestoreOnAction().
+									equalsIgnoreCase(env.getProperty(ResidentConstants.VID_ACTIVE_STATUS))){
+						if(getNumberOfPerpetualVidFromUin(uin).getT2().equals(individualId)){
```

Comment:
Merge this if condition with previous if condition.

Reply:
done.